cpg_workflows/stages/rd_combiner.py

```diff
@@ -652,7 +732,22 @@ def expected_outputs(self, dataset: Dataset) -> Path:
         """
         Expected to generate a matrix table
         """
-        return dataset.prefix() / 'mt' / self.name / f'{get_workflow().output_version}-{dataset.name}.mt'
+        if only_families := config_retrieve(['workflow', dataset.name, 'only_families'], []):
+            family_sg_data = query_for_family_sequencing_groups(
+                dataset.name,
+                config_retrieve(['workflow', 'sequencing_type']),
+                only_families,
+            )
+            n_sgs = len(family_sg_data['family_sg_ids'])
+            n_families = len(family_sg_data['only_family_ids'])
+            return (
+                dataset.prefix()
+                / 'mt'
+                / self.name
+                / f'{get_workflow().output_version}-{dataset.name}-{n_sgs}_sgs-{n_families}_families.mt'
+            )
```

Comment:
If we carry out this same process twice, each for a different trio from the same callset, the `{dataset.name}-{n_sgs}_sgs-{n_families}_families.mt` portion of the filename would be identical, so we wouldn't repeat the Stage.


Can we incorporate a partial hash of the CPG IDs in this to make sure we're keeping these outputs in parallel? Borrowing from [here](https://github.com/populationgenomics/production-pipelines/blob/20bfbf9b60dde97b942891bc96e4a5bde43328e7/cpg_workflows/targets.py#L82-L91) we could do

```suggestion
            import hashlib
            sg_ids = family_sg_data['family_sg_ids']
            h = hashlib.sha256(''.join(sorted(sg_ids).encode()).hexdigest()[:4]
            n_sgs = len(sg_ids)
            n_families = len(sg_ids)
            return (
                dataset.prefix()
                / 'mt'
                / self.name
                / f'{get_workflow().output_version}-{dataset.name}-{n_sgs}_sgs-{n_families}_{h}_families.mt'
            )
```